src/wheel/util.py

```diff
@@ -30,3 +31,14 @@ def as_bytes(s):
         return s.encode("utf-8")
     else:
         return s
+
+
+def log(msg, *, error=False):
```

Comment:
Rather than implementing a custom logger, I'd prefer to re-use Python's `logging` module. Does that not work in this scenario because the logger is not initialized?  :/

Reply:
Indeed. Even setuptools still uses the `distutils` logger internally.

Reply:
I think this is a defect that needs to be addressed in Setuptools. It should supply a `log` object similar to what distutils does (but probably using Python's logging module).

Reply:
Ok, but until that's in place, do you see any problem with using this approach now? It should be functionally equivalent to the distutils logging code (just with a newer syntax and leveraging `print()`.

Reply:
In pypa/setuptools#2973, I've captured the need for a replacement. I realize that Setuptools won't be able to supply a replacement for distutils in such a way that it's available on older Setuptools releases, so wheel will probably not be able to rely on it for a while. Therefore, I'd recommend that wheel provide an interface-compatible fallback, and perhaps just a copy of the implementation that makes it into Setuptools. I may be able to get that implementation out today.

Reply:
Great, looking forward to it.

Reply:
In pypa/setuptools#2974, I believe I have a technique that will configure the Python logging mechanism to log to stderr/stdout as distutils.log currently does.

Then, in https://github.com/pypa/wheel/tree/python-logging, I illustrate how this approach could be employed in wheel to log through that mechanism. I could have even restored the `logger` name and `%` substitutions, making the diff from main very small, but opted to leave those changes.

The problem with that approach is that for older Setuptools, nothing will have configured the logger, so the default configuration of the logger will apply. In https://github.com/pypa/wheel/tree/python-logging-compat, I apply another commit that copies the code that configures the logger and run that. That approach will give a compatible effect, but still won't honor any calls to `distutils.log.set_threshold` on older Setuptools. That seems like an acceptable tradeoff (and one this PR has already made).

Please have a look at the Setuptools PR and then consider pulling one or both of the aforementioned commits for this PR.

Reply:
Thanks! I'll look at those branches later.